pages/DSNP/DSNP-Announce.md

```diff
@@ -0,0 +1,102 @@
+# DSNP Announce
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Draft |
+
+## Purpose
+1. Describe the method to announce [DSNP Messages](/DSNP/DSNP-Messages) to the blockchain.
+1. Provide Solidity code examples (pending final contract)
+1. Facilitate use of SDK and locating of on-chain data
+
+## Assumptions
+* Chain messages are on Ethereum
+* Message data is posted via [Ethereum log events](https://medium.com/mycrypto/understanding-event-logs-on-the-ethereum-blockchain-f4ae7ba50378)
+* Signature algorithm is [secp256k1](https://en.bitcoin.it/wiki/Secp256k1). This allows the use `ecreover`
+  to get public keys. A public key also need not be included in a log event for ease of validation.
+* content hashes are created via the same [keccak-256 hashing algorithm](https://en.wikipedia.org/wiki/SHA-3) used by Ethereum.
+
+## DSNP Announce
+
+### DSNPMessage Event
+
+The DSNPMmessage event is a standard event topic to be used for all DSNP Messages.
+
+```solidity
+interface DSNPAnnounce {
+    event DSNPMessage(bytes32 indexed dsnpRoute, uint16 dsnpType, bytes[] dsnpData);
+}
+```
+
+| field | description | type |
+|-------|-------------|------|
+| dsnpRoute | Route target | bytes32 indexed |
+| dsnpType | Enum from [DSNP Message Types](/DSNP/DSNP-Message-Types) | unit16 |
+| dsnpData | Matching serialized data from [DSNP Messages](/DSNP/DSNP-Messages) | bytes |
+
+### DSNP Route Field
+
+The route target is a filter that enables smaller applications to filter the large mass of DSNP Messages down to a smaller dataset.
+The route is a 32 byte value, most often the id of the sender, but can also be the id of a message for replies and reactions.
+
+```typescript
+const dsnpTopic = keccak256("DSNPMessage(bytes32 indexed dsnpRoute, uint16 dsnpType, bytes");
+
+// Subscribe to users
+const usersContent = web3.eth.subscribe('logs', {
+    topics: [[dsnpTopic, idOfUser0], [dsnpTopic, idOfUser1]]
+}, userContentHandler);
+
+// Get past logs to find replies to a message id
+const replies = web3.eth.getPastLogs({
+  fromBlock: message.blockNumber,
+  topics: [[dsnpTopic, message.dsnpId]]
+}, replyReactionHandler);
+```
+
+#### Route Field Example: From Address
+
+This is the most common route.
+The identity of the sender.
+Used for broadcasting messages as well as sending notice of a profile update.
+If you want to read messages from this user, look for this route.
+
+#### Route Field Example: Message Id
+
+The id of a message the new message is related to.
+Used for replies, reactions, and other message related metadata.
+If you want to gather additional messages about a message you are showing, look for this route.
+
+#### Route Field Example: Personal Id
+
+Messages that are for a user and their applications only.
+Usually contains encrypted data.
+Generated by taking the keccak-256 hash of the user's id.
```

Comment:
I'm not sure what I think of this still. Both as a concept and this implementation. Also I'm not sure about the list of things I applied to it.

I like the idea of a `hash(sender's id)` meaning that personal stuff is less obviously connected. It is NOT private however, so perhaps that is just not worth the work?

Another option is just to prepend a 1 to the sender's id (only works when the sender's id is < 32 bytes which it is currently).
Another option is to just add 1 to the sender's id (only works if the ids are in a large space, which they are currently).